crates/bitwarden-error-macro/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+mod args;
+mod attribute;
+mod basic;
+mod flat;
+mod full;
+
+/// A procedural macro for generating error types with customizable serialization behavior.
+///
+/// # Attributes
+///
+/// ## Error type
+
+/// - `basic`: The error is converted into a string using the `ToString` trait.
+/// - `flat`: The error is converted into a flat structure using the `FlatError` trait.
+/// - `full`: The entire error stack is made available using `serde`.
```

Comment:
Docs look pretty nice! It would be good to have some recommendations on when to use some variant over the others. As a new developer it might be hard to know when each is appropiate.

Reply:
I was thinking about writing some contributing docs page about this, but honestly people will just look at how other's code looks and copy paste :D